### PR TITLE
implemented `read_only` node setting

### PR DESCRIPTION
**Pull request by seut**
Added a node setting `node.sql.read_only`. If set to true,
only SQL statements resulting in read operations are allowed.

### CODE REVIEW

File: sql/src/main/java/io/crate/exceptions/ForbiddenException.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public ForbiddenException(String s) {
+        super(s);
+    }
+
+    public ForbiddenException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    @Override
+    public int errorCode() {
```
Comment (dobe) [inline sql/src/main/java/io/crate/exceptions/ForbiddenException.java]: would probably make sense to make this abstract?

File: sql/src/test/java/io/crate/integrationtests/ReadOnlyNodeIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void testWriteOperationThrowsException() throws Exception {
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("Only read operations are allowed on this node");
+
+        execute("create table test (id int)");
+    }
+
```
Comment (dobe) [inline sql/src/test/java/io/crate/integrationtests/ReadOnlyNodeIntegrationTest.java]: can you write a test with insert from query. in order to check nested statements

File: sql/src/main/java/io/crate/exceptions/ForbiddenException.java (hunk 1 above)
Comment (seut) [inline sql/src/main/java/io/crate/exceptions/ForbiddenException.java]: as long as we don't need a generic unspecific forbidden exception, which we currently don't need, yes. so yes ;)

File: sql/src/test/java/io/crate/integrationtests/ReadOnlyNodeIntegrationTest.java (hunk 2 above)
Comment (dobe) [inline sql/src/test/java/io/crate/integrationtests/ReadOnlyNodeIntegrationTest.java]: and probably also a test with explain

**Comment by dobe** [comment]
ack

**Pull request merged**